controller/manager/manager.go

```diff
@@ -0,0 +1,101 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"fmt"
+
+	operatorv1alpha1 "github.com/projectcontour/contour-operator/api/v1alpha1"
+	contourcontroller "github.com/projectcontour/contour-operator/controller/contour"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var (
+	scheme = runtime.NewScheme()
+	mgrLog = ctrl.Log.WithName("manager")
+)
+
+func init() {
+	_ = clientgoscheme.AddToScheme(scheme)
+	_ = operatorv1alpha1.AddToScheme(scheme)
+}
+
+// NewContourManager creates a Manager with options specified by opts with
+// a Reconciler configured by cfg that reconciles Contour objects.
+func NewContourManager(opts ctrl.Options, cfg contourcontroller.Config) (manager.Manager, error) {
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), opts)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create contour manager: %w", err)
+	}
+
+	reconciler := newContourReconciler(mgr, cfg)
+
+	if err = builder.
+		ControllerManagedBy(mgr).
+		For(&operatorv1alpha1.Contour{}).
+		Watches(&source.Kind{Type: &appsv1.Deployment{}}, enqueueRequestForOwningContour()).
+		Watches(&source.Kind{Type: &appsv1.DaemonSet{}}, enqueueRequestForOwningContour()).
```

Comment:
We didn't have these watches before, right?  These watches mean not only that the operator updates the CR's status in response to updates to the operands, but also that the operator reconciles and reverts changes to the operands by external agents.  It is a little surprising to see in a change related to status, so I wanted to call it out, but it is desirable behavior, so might as well go ahead and do it now.

Reply:
> We didn't have these watches before, right?

That's correct.

> It is a little surprising to see in a change related to status, so I wanted to call it out, but it is desirable behavior, so might as well go ahead and do it now.

I thought the controller needs these watches so it can update the status of `Contour`. In my testing, the status of `Contour` does not get updated without these watches.

Reply:
> > It is a little surprising to see in a change related to status, so I wanted to call it out, but it is desirable behavior, so might as well go ahead and do it now.
> 
> I thought the controller needs these watches so it can update the status of `Contour`. In my testing, the status of `Contour` does not get updated without these watches.

Right, sorry I wasn't more clear.  These changes are necessary for proper status updating (that's the intended main effect).  They have a side effect as well, namely that they cause the operator to reconcile modifications to the resources by others.  The main effect and side effect are both desirable; I just want to call out the side effect as noteworthy.